frontend/src/lib/stores/stores.ts

```diff
@@ -162,11 +163,21 @@ export const election: Readable<ElectionProps | undefined> = derived(
 );
 
 /**
- * Utility store for candidates as part of `PageData`.
+ * Utility store for candidates as part of `PageData` that also handles possible filtering by answers and sorting of candidates.
  */
 export const candidates: Readable<Promise<CandidateProps[]>> = derived(
   page,
-  ($page) => $page.data.candidates?.then((d) => d.sort(sortCandidates)) ?? Promise.resolve([]),
+  ($page) => {
+    const candsPromise = $page.data.candidates;
+    if (!candsPromise) return Promise.resolve([]);
+    // Technically, we should derive this store also from settings and opinionQuestions, but this will cause unnecessary updates and these are all dependent on page, so we just fetch them manually
+    const hideIfMissing = get(settings).entities?.hideIfMissingAnswers?.candidate;
```

Comment:
In theory, we could handle this in the backend but it would need it's own route or controller logic.